### PR TITLE
Disable view survey if validation errors

### PR DESCRIPTION
**Pull request by heofs**
### What is the context of this PR?
<https://trello.com/c/S4FNVqG1/1096-disabling-view-survey-4>
Added a function that checks if there is any validation errors and disables the view survey button if any errors are found. 

### How to review 
1. Create a new questionnaire
1. Button should be immediately disabled because you haven't added any questions.
1. Add question text and blur out of the field, the button should be enabled again.
1. Try creating new answers and questions and see that the behaviour is consistent.

### CODE REVIEW

File: eq-author-api/schema/resolvers/base.js [hunk 1]
```diff
@@ -514,6 +514,8 @@ const Resolvers = {
       }
       return "Read";
     },
+    totalErrorCount: (questionnaire, args, ctx) =>
+      ctx.validationErrorInfo ? ctx.validationErrorInfo.totalCount : 0,
```
Comment (tgandrews) [inline eq-author-api/schema/resolvers/base.js]: Why is this check necessary?

Comment (heofs) [reply eq-author-api/schema/resolvers/base.js]: The 3 tests under `duplicate a questionnaire` in `eq-author-api/schema/tests/duplication.test.js` was failing without it. Giving totalCount is undefined.

Comment (tgandrews) [reply eq-author-api/schema/resolvers/base.js]: I guess this is an issue with duplication not validating the questionnaire

File: eq-author/src/components/EditorLayout/Header/index.test.js [hunk 2]
```diff
@@ -92,14 +93,33 @@ describe("Header", () => {
...
+
+      const viewSurveyButton = getByTestId("btn-preview");
+      expect(viewSurveyButton.getAttribute("disabled")).toBe("");
+    });
+
+    it("should not disable the view survey button when questionnaire is valid", () => {
+      const { getByTestId } = renderWithContext(<Header {...props} />);
+
+      const viewSurveyButton = getByTestId("btn-preview");
+      expect(viewSurveyButton.getAttribute("disabled")).toBeFalsy();
```
Comment (tgandrews) [inline eq-author/src/components/EditorLayout/Header/index.test.js]: Empty string is also falsy. So we can check for something explicitly like `null` or `undefined`

File: eq-author-api/schema/resolvers/base.js [hunk 3]
```diff
@@ -514,6 +514,8 @@ const Resolvers = {
       }
       return "Read";
     },
+    totalErrorCount: (questionnaire, args, ctx) =>
```
Comment (tgandrews) [inline eq-author-api/schema/resolvers/base.js]: Can we add a test on the server api for this.

File: eq-author/src/components/EditorLayout/Header/index.test.js (hunk 2 above)
Comment (heofs) [reply eq-author/src/components/EditorLayout/Header/index.test.js]: Aah thats right, I changed it to check for `null`.

File: eq-author-api/schema/resolvers/base.js (hunk 3 above)
Comment (heofs) [reply eq-author-api/schema/resolvers/base.js]: doesnt `totalErrorCount: expect.any(Number),` in `eq-author-api/schema/tests/questionnaire.test.js` test this properly?

Comment (tgandrews) [reply eq-author-api/schema/resolvers/base.js]: Good point I missed that.

File: eq-author/src/components/EditorLayout/Header/index.test.js (hunk 2 above)
Comment (SamGodwin2) [reply eq-author/src/components/EditorLayout/Header/index.test.js]: Might it be worth bringing in something like https://github.com/testing-library/jest-dom to assert the button is disabled. This way of doing it doesn't feel very rtl-ly if that makes sense?

Comment (tgandrews) [reply eq-author/src/components/EditorLayout/Header/index.test.js]: That sounds like a much better idea. Good find.

Comment (heofs) [reply eq-author/src/components/EditorLayout/Header/index.test.js]: Now checks using this. And fixed the button so you no longer can tab into it.

Comment (tgandrews) [reply eq-author/src/components/EditorLayout/Header/index.test.js]: Whats wrong with being able to tab to it? https://ux.stackexchange.com/questions/103239/should-disabled-elements-be-focusable-for-accessibility-purposes

Comment (heofs) [reply eq-author/src/components/EditorLayout/Header/index.test.js]: The user was able to tab into it and click enter to still go to the link, even when it was disabled. Hamish suggests not being able to tab into it. And there is no point being able to tab into something you cant interact with.

File: eq-author/src/components/EditorLayout/Header/index.test.js [hunk 4]
```diff
@@ -111,14 +111,14 @@ describe("Header", () => {
       const { getByTestId } = renderWithContext(<Header {...props} />);
 
       const viewSurveyButton = getByTestId("btn-preview");
-      expect(viewSurveyButton.getAttribute("disabled")).toBe("");
+      expect(viewSurveyButton).toHaveAttribute("disabled");
```
Comment (tgandrews) [inline eq-author/src/components/EditorLayout/Header/index.test.js]: You can use `toBeDisabled`

**Pull request merged**